apps/studio/components/interfaces/Auth/AuthProvidersForm/FormField.tsx

```diff
@@ -1,288 +1,276 @@
-import dayjs from 'dayjs'
+import { useState } from 'react'
 import { Eye, EyeOff } from 'lucide-react'
-import { useEffect, useState } from 'react'
-import ReactMarkdown from 'react-markdown'
+import type { Control } from 'react-hook-form'
+import dayjs from 'dayjs'
 
 import { Markdown } from 'components/interfaces/Markdown'
 import { DatePicker } from 'components/ui/DatePicker'
 import { BASE_PATH } from 'lib/constants'
-import { Button, Input, InputNumber, Listbox, Toggle } from 'ui'
-import type { Enum } from './AuthProvidersForm.types'
+import {
+  Button,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Select_Shadcn_,
+  SelectContent_Shadcn_,
+  SelectItem_Shadcn_,
+  SelectTrigger_Shadcn_,
+  SelectValue_Shadcn_,
+  TextArea_Shadcn_,
+  Switch,
+} from 'ui'
+import { Input } from 'ui-patterns/DataInputs/Input'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import type { Provider } from './AuthProvidersFormValidation'
 
 interface FormFieldProps {
   name: string
-  properties: any
-  formValues: any
-  setFieldValue: (field: string, v: any) => any
+  properties: Provider['properties'][string]
+  control: Control
   disabled?: boolean
 }
 
 function formatDate(date: Date): string {
   return dayjs(date).format('dddd, MMMM D, YYYY HH:mm:ss Z')
 }
 
-const FormField = ({
-  name,
-  properties,
-  formValues,
-  disabled = false,
-  setFieldValue,
-}: FormFieldProps) => {
-  const [hidden, setHidden] = useState(!!properties.isSecret)
-  const [dateAsText, setDateAsText] = useState(
-    formValues[name] ? formatDate(new Date(formValues[name])) : ''
-  )
-
-  useEffect(() => {
-    if (properties.show && properties.show.key && !formValues[properties.show.key]) {
-      setFieldValue(name, '')
-      setDateAsText('')
-    }
-  }, [properties.show && properties.show.key && !formValues[properties.show.key]])
-
-  if (properties.show) {
-    if (properties.show.matches) {
-      if (!properties.show.matches.includes(formValues[properties.show.key])) {
-        return null
-      }
-    } else if (!formValues[properties.show.key]) {
-      return null
-    }
-  }
```

Comment:
opted to lift this out of the field itself and into the `ProviderForm` component to simplify the props